lib/statsig/logging.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Statsig.Logging do
+  use GenServer
+  require Logger
+
+  @type state :: %{
+    events: list(),
+    flush_timer: reference() | nil
+  }
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{
+      events: [],
+      flush_timer: nil,
+    }, name: __MODULE__)
+end
+
+  def flush do
+    GenServer.call(__MODULE__, :flush)
+  end
+
+  @impl true
+  def init(state) do
+    timer = Process.send_after(__MODULE__, :flush_events, get_flush_interval())
+        {:ok, Map.put(state, :flush_timer, timer)}
+  end
+
+  @impl true
+  def handle_call(:flush, _from, state) do
+    new_state = flush_events(state)
+    unsent_events = new_state.events
+    {:reply, unsent_events, new_state}
+  end
+
+  @impl true
+  def handle_info(:flush_events, state) do
+    timer = Process.send_after(__MODULE__, :flush_events, get_flush_interval())
+
+    new_state = flush_events(Map.put(state, :flush_timer, timer))
+    {:noreply, new_state}
+  end
+
+  @impl true
+  def handle_cast({:log_event, event}, state) do
+    new_events = [event | state.events]
+    new_state = %{state | events: new_events}
+    {:noreply, new_state}
+  end
+
+  @impl true
+  def terminate(reason, state) do
+    if length(state.events) > 0 do
```

Comment:
i feel like i commented on this, what happened? 
The comment is that flush_event should just get called and it should handle the empty case.

Reply:
I hadn't finished going through the feedback - I went through the application and configs, but not logging yet :p